test/widgets/subscription_list_test.dart

```diff
@@ -89,18 +89,71 @@ void main() {
     check(isUnpinnedHeaderInTree()).isTrue();
   });
 
-  testWidgets('subscription sort', (tester) async {
+  testWidgets('pinned subscriptions are sorted', (tester) async {
     await setupStreamListPage(tester, subscriptions: [
       eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: true),
-      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: true),
       eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: true),
+    ]);
+    check(isPinnedHeaderInTree()).isTrue();
+
+    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
+    check(streamListItems.map((e) => e.subscription.streamId)).deepEquals([4, 3, 2, 1]);
+  });
+
+  testWidgets('pinned subscriptions are sorted case insensitively', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1, name: 'D'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 3, name: 'B'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: true),
+    ]);
+    check(isPinnedHeaderInTree()).isTrue();
+
+    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
+    check(streamListItems.map((e) => e.subscription.streamId)).deepEquals([4, 3, 2, 1]);
+  });
+
+  testWidgets('unpinned subscriptions are sorted', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: false),
       eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: false),
     ]);
+    check(isUnpinnedHeaderInTree()).isTrue();
+
+    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
+    check(streamListItems.map((e) => e.subscription.streamId)).deepEquals([4, 3, 2, 1]);
```

Comment:
When making a larger number of test cases, one aspect that's key for keeping it easy to understand the tests and maintain them is to keep each test case short — focused and concise, with only the details that are important to making that test case work.

So in particular, now that each of these test cases is only testing one thing, they don't need four items in the list — I think two items will make the point just as well.

Or three: like b, a, c -> a, b, c, to ensure that the test doesn't pass if we end up with a bug where we're forgetting to sort but happen to show things in reverse order relative to the list in the input data.